ai_presenter/text_ai/chat/messages.py

```diff
@@ -0,0 +1,60 @@
+import tiktoken
+
+# the absolute maximum number of tokens that
+# can be used in a message for a chatgpt api call
+MAX_TOKEN_LIMIT = 4096
+
+# the number of tokens allowed in a message for a chatgpt api call
+# before old scenes start getting deleted in order to keep addition
+# of new scenes from going over the MAX_TOKEN_LIMIT
+SOFT_TOKEN_LIMIT = 2500
+
+
+class Messages:
+    def __init__(self, gpt_initializer):
+        self.gpt_setup = gpt_initializer
+        self.scenes = []
+        self.requests = []
+        self.responses = []
+        self.max_token_limit = MAX_TOKEN_LIMIT
+        self.soft_token_limit = SOFT_TOKEN_LIMIT
+
+# scene expected in format of:
+# {"role": "user"/"assistant", "content": json.dumps(self.user_message)}
+    def update_scenes(self, scene: dict):
+        self.token_check()
+
+        if scene['role'] == 'assistant':
+            self.responses.append(scene)
+            self.scenes.append(scene)
```

Comment:
Remove `self.scenes` and have `get_scene_count` return the added length of both.